near-sdk-macros/src/core_impl/utils/mod.rs

```diff
@@ -152,6 +173,29 @@ pub struct SanitizeSelfResult {
     pub self_occurrences: Vec<Span>,
 }
 
+pub fn get_error_type_from_status(status_result: &StatusResult) -> syn::Type {
+    let ty = status_result.result_type.clone();
+    syn::parse_quote! { <#ty as near_sdk::__private::ResultTypeExt>::Error }
+}
+
+pub fn standardized_error_panic_tokens(error_type: &syn::Type) -> TokenStream2 {
+    quote! {
+        // Initial error is wrapped into a struct to be able to serialize the type of it.
+        #[near(serializers=[json])]
+        struct ErrorWrapper {
+            error_type: &'static str,
+            value: #error_type,
+        };
+
+        let err = ErrorWrapper {
+            error_type: std::any::type_name::<#error_type>(),
+            value: err,
+        };
+        let err = ::near_sdk::serde_json::json!{{"error": err}};
```

Comment:
Let's copy the errors structure from NEAR RPC: https://docs.near.org/api/rpc/contracts#what-could-go-wrong

```json
{
  "error": {
    "name": <ERROR_TYPE>,
    "cause": {
      "info": {..},
      "name": <ERROR_CAUSE>
    }
  }
}
```

For example, for user-generated errors:

```json
{
  "error": {
    "name": "CUSTOM_CONTRACT_ERROR",
    "cause": {
      "info": {..},
      "name": "MY_ERROR_TYPE"
    }
  }
}
```

For SDK-generated errors:

```json
{
  "error": {
    "name": "SDK_CONTRACT_ERROR",
    "cause": {
      "info": {..},
      "name": "INVALID_INPUT" // "UNEXPECTED_ATTACHED_BALANCE" | "INVALID_INTERNAL_STATE" | ...
    }
  }
}
```